include/libtorrent/utp_stream.hpp

```diff
@@ -300,12 +299,6 @@ struct TORRENT_EXTRA_EXPORT utp_stream
 	template <class Handler>
 	void async_connect(endpoint_type const& endpoint, Handler const& handler)
 	{
-		if (!is_v4(endpoint))
-		{
-			m_io_service.post(std::bind<void>(handler, boost::asio::error::operation_not_supported));
-			return;
-		}
```

Comment:
@ssiloti I was surprised to find this. I can't recall any good reason to block IPv6 uTP connections, can you?
When enabling IPv6 and the commented out simulation, it failed because the path MTU discovery didn't work (which I fixed in the sim and updated below)

Reply:
No, we should definitely be able to support uTP over IPv6. This code is very old (committed in November of 2010) so I can only assume it is a vestigial workaround for some hopefully long since resolved issue.